File: Interview Practice/Arrays/firstNotRepeatingCharacter/first_not_repeating_character.js
function firstNotRepeatingCharacter(s) {
	var appeared = new Array(26);
	appeared.fill(0);

	for (var i = 0; i < s.length; i++) {
		appeared[s[i].charCodeAt(0) - 'a'.charCodeAt(0)]++;
	}

	for (var i = 0; i < s.length; i++) {
		if (appeared[s[i].charCodeAt(0) - 'a'.charCodeAt(0)] == 1) {
			return s[i];
		}
	}

	return '_';
}

File: Interview Practice/Arrays/firstNotRepeatingCharacter/first_not_repeating_character.java
char firstNotRepeatingCharacter(String s) {
	int[] appeared = new int[26];
	Arrays.fill(appeared, 0);

	for (int i = 0; i < s.length(); i++) {
		appeared[s.charAt(i) - 'a']++;
	}

	for (int i = 0; i < s.length(); i++) {
		if (appeared[s.charAt(i) - 'a'] == 1) {
			return s.charAt(i);
		}
	}

	return '_';
}